docs/tutorials/custom/custom-base-dog.md

```diff
@@ -345,42 +345,58 @@ Name your module `my-custom-base`.
+![Screenshot of the Viam app Config tab with the Modules subtab open, showing my-custom-base configured.](/tutorials/custom-base-dog/module-config.png)
 
 ## Configure the components
 
 Now that the custom base code is set up, you need to configure all your hardware components.
-Navigate to the **Components** subtab of the **Config** tab.
+Navigate to the **Components** subtab of your robot's **Config** tab.
 
-### Configure the base
+### Configure the camera
 
-In the **Create Component** field, give your base a name.
-We called ours "quadruped".
-In the **Type** drop-down select `base`.
-In the **Model** field, type in `viamlabs:base:robotdog`.
-Click **Create Component**.
+Configure the ribbon camera on the dog as a `webcam` following our [webcam documentation](/components/camera/webcam/).
 
-![Screenshot of the Viam app CONFIG tab with the Config subtab open, showing quadruped configured.](/tutorials/custom-base-dog/config-modular-component.png)
+Click **Save config**.
 
-You need to add the `ip_address` and `port` attributes to your base config.
-In the attributes field, paste the following, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`):
+### Configure the base
+
+Because your custom base relies on a local module, you need to use raw JSON to configure your modular resource.
+Use the **Mode** selector in the upper-left corner of the **Config** tab to switch to **Raw JSON** mode.
+
+Locate the `"components": []` portion of the config file.
+At the end of your camera configuration, add a comma and then add the following base configuration:
 
 ```json
-{
-  "ip_address": "<HOSTNAME>.local",
-  "port": 5001
-}
+    {
+      "namespace": "rdk",
+      "name": "quadruped",
+      "type": "base",
+      "model": "viamlabs:base:robotdog",
+      "attributes": {
+          "ip_address": "<HOSTNAME>.local",
+        "port": 5001
+      },
+      "depends_on": []
+    }
 ```
 
+Edit the `ip_address` attribute to match your robot's IP address, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`).
 If this doesn't work, you can instead try using the IP address of the machine where the module is running, for example, `"ip_address": "10.0.0.123"`.
 
+If you are using a port other than `5001`, edit the `port` attribute.
 `5001` is the default port for sending and receiving instructions to and from the Freenove server.
 
-Click **Save Config**.
+Click **Save config**.
 
-### Configure the camera
+Your raw JSON configuration should look similar to the following:
 
-Configure the ribbon camera on the dog as a `webcam` following our [webcam documentation](/components/camera/webcam/).
+![Raw JSON mode on the Config tab, showing the components and modules sections of the robot's raw JSON config.](/tutorials/custom-base-dog/raw-json.png)
+
+Toggle back to **Builder** mode and make sure a card has been generated:

```

Comment:
nit: is "card" the right word here? I've been using "pane" or "configuration screen". No preference, just haven't seen "card" before.

Reply:
```suggestion
Toggle back to **Builder** mode and make sure a configuration panel has been generated:
```
I don't even know; I've been using "panel" as well--I don't think we've totally standardized. TBH I like card but searching the docs I see that it's only used in the navigation service config instructions, so I won't use it.